components/api/docker_registry/registry.bal

```diff
@@ -0,0 +1,129 @@
+// ------------------------------------------------------------------------
+//
+// Copyright 2019 WSO2, Inc. (http://wso2.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+//
+// ------------------------------------------------------------------------
+
+import ballerina/config;
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+
+http:Client dockerRegistryClientEP = new(config:getAsString("docker.registry.url"), config = {
+    secureSocket: {
+        trustStore: {
+            path: config:getAsString("security.truststore"),
+            password: config:getAsString("security.truststorepass")
+        },
+        verifyHostname: false
+    }
+});
+
+public function getResponseFromManifestAPI(string orgName, string imageName, string artifactVersion = "", string digest = "", string bearerToken = "")
```

Comment:
This method seems like it is doing two completely different things. This should be two separate methods ideally.